p2-maven-plugin/src/main/java/org/eclipse/tycho/p2maven/SlicingOptions.java

```diff
@@ -0,0 +1,100 @@
+/*******************************************************************************
+ * Copyright (c) 2009, 2017 IBM Corporation and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.tycho.p2maven;
+
+import java.util.Hashtable;
+import java.util.Map;
+
+/**
+ * Holding the slicing option for a permissive slicer, the defaults are:
+ * <ul>
+ * <li>includeOptionalDependencies = true</li>
+ * <li>everythingGreedy = true</li>
+ * <li>forceFilterTo = true</li>
+ * <li>considerStrictDependencyOnly = false</li>
+ * <li>followOnlyFilteredRequirements = false</li>
+ * <li>filter = null</li>
```

Comment:
These options are not very clear to me. Could you add some description of what they're supposed to do? I also don't know how I could configure a HashMap (`filter`) in a `pom.xml`

Reply:
Yes these are rather special options I think if the mojo is really used more one would want a dedicated description on it.

Reply:
I enhanced the description now!